<launch>

    <!-- Args for the Agent Nodes -->
    <arg name="serial_id_1"                         default="agent_1" />
    <arg name="x_1"                                 default="2" />
    <arg name="y_1"                                 default="0" />
    <arg name="theta_1"                             default="0" />

    <arg name="serial_id_2"                         default="agent_2" />
    <arg name="x_2"                                 default="0" />
    <arg name="y_2"                                 default="3" />
    <arg name="theta_2"                             default="0" />

    <!-- <arg name="serial_id_3"                         default="agent_3" />
    <arg name="x_3"                                 default="7" />
    <arg name="y_3"                                 default="8" />
    <arg name="theta_3"                             default="45" /> -->

    <!-- <arg name="serial_id_4"                         default="agent_4" />
    <arg name="x_4"                                 default="8" />
    <arg name="y_4"                                 default="3" />
    <arg name="theta_4"                             default="45" /> -->

    <node
        type="rviz"
        name="rviz"
        pkg="rviz"
        args="-d $(find multi_agent_planner)/rviz/rviz_config.rviz" />

    <!-- Motion Planner Node -->
    <node
        type="motion_planner_node"
        name="motion_planner_node"
        pkg="multi_agent_planner"
        output="screen" />

    <!-- Agent 1 Node -->
    <node
        type="agent_node"
        name="$(arg serial_id_1)_node"
        pkg="multi_agent_planner"
        ns="$(arg serial_id_1)"
        args="$(arg serial_id_1) $(arg x_1) $(arg y_1) $(arg theta_1)"
        output="screen" />

    <!-- Agent 2 Node -->
    <node
        type="agent_node"
        name="$(arg serial_id_2)_node"
        pkg="multi_agent_planner"
        ns="$(arg serial_id_2)"
        args="$(arg serial_id_2) $(arg x_2) $(arg y_2) $(arg theta_2)"
        output="screen" />

    <!-- Agent 3 Node (for fun)-->
    <!-- <node
        type="agent_node"
        name="$(arg serial_id_3)_node"
        pkg="multi_agent_planner"
        ns="$(arg serial_id_3)"
        args="$(arg serial_id_3) $(arg x_3) $(arg y_3) $(arg theta_3)"
        output="screen" /> -->

    <!-- Agent 4 Node (for fun)-->
    <!-- <node
        type="agent_node"
        name="$(arg serial_id_4)_node"
        pkg="multi_agent_planner"
        ns="$(arg serial_id_4)"
        args="$(arg serial_id_4) $(arg x_4) $(arg y_4) $(arg theta_4)"
        output="screen" /> -->

</launch>
